src/secure-proxy-handler.ts

```diff
@@ -1,14 +1,15 @@
 import {
     apply,
+    assign,
     construct,
     isUndefined,
     ObjectDefineProperty,
+    defineProperty,
```

Comment:
Here I make a distinction between `Reflect.defineProperty` and `ObjectDefineProperty`. The important bit is if setting the property fails `Reflect.defineProperty` (which we use in most places) won't error. But sometimes like in a proxy trap you want the error to be thrown which is why I kept `ObjectDefineProperty` around.